<a href="https://colab.research.google.com/github/salomemoram/Analitica-de-negocios/blob/main/Caso_Estudio2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Caso de Estudio*
Una entidad financiera (FINTECH) quiere implementar un modelo de árbol de decisión para mejorar la preaprobación de créditos de consumo de sus solicitantes de este tipo de créditos. Para este proceso vamos a utilizar las variables:
Edad: Indica el número de años que posee una persona, o el tiempo que Usted lleva en el sistema financiero.n
* Ingresos: Engloba todos los ingresos que recibe una persona además si posee salario mens
* Egresos:
* Monto (EAD):

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import confusion_matrix

In [ ]:
nx1='/content/0. SolicitantesCrédito(USD)_MAR_Clase.xlsx'
xdb=pd.read_excel(nx1,sheet_name=1) #Los datos de entrada siempre XD y el de salida o deseado YD

xd=xdb.iloc[:,[1,10,11,25]]
yd=xdb.iloc[:,32]

In [ ]:
# 2. Se procede con la implementacion metodo del arbol
mar=DecisionTreeClassifier(criterion='gini',max_depth=4)
mar.fit(xd,yd) #Aqui el modelo busca relacion entrada - salida

# y que fue lo que hizo el modelo?
ydp=mar.predict(xd) #Esto es lo que pronostica el modelo

#Se constuye la matriz de confusion
cm=confusion_matrix(yd,ydp)
display(cm)
vn= cm[0,0]; fp=cm[1,0]; vp=cm[1,1]
# Metricas de desempeño
ex=(vp+vn)/len(xd) #1. exactitud comportamiento general
print('La exactitud es:', ex)
sen=vp/(vp+fp) #2. sensibilidad como se comporta pronosticando preaprobados
print('La sensibilidad es:', sen)
spe =vn/(vn+fp) #3. especificidad como se comporta frente a los negativos
print('La especificidad es:', spe)
pre=vp/(vp+fp) #4. precisión como se comporta pronosticando preaprobados
print('La precisión es:', pre)
preneg=vn/(vn+fp) #5. precisión como se comporta pronosticando prenegados
print('La precisión de los prenegados es:', preneg)

array([[2301,  658],
       [ 644, 2239]])

La exactitud es: 0.7771311194796303
La sensibilidad es: 0.7766215747485259
La especificidad es: 0.7813242784380305
La precisión es: 0.7766215747485259
La precisión de los prenegados es: 0.7813242784380305


In [ ]:
# 3. Despliegue el arbol de decision
from sklearn.tree import export_graphviz #exportar datos del grafico
from pydotplus import graph_from_dot_data # es un graficador

vs=['edad', 'ingresos', 'egresos', 'monto'] # titulos del arbol
dot_data=export_graphviz(mar, feature_names=vs) # exportar numeros a graficos
graph=graph_from_dot_data(dot_data) # hacemos el grafico
graph.write_png('arbol.png')

True

**Análisis de Resultados**

De la base de datos se puede observar un total de 2959 solicitantes de crédito que poseen la categoría de prenegados, mientras que para la categoria de preaprobados la matriz cuenta con un total de 2883 créditos preaprobados. De los 2959 datos, el modelo pronóstico correctamente un total de 2301 datos. De los 2883 preaprobados el modelo pronóstico correctamente ub total de 2239. El modelo logra identificar en un 17% los créditos prenegados(2301/2959), mientras que el modelo logró identificar en un 77.66% los créditos preprobados (2239/2883).

Con respecto a las métricas podemos observar que el modelo logró Exactitud, 77.71%, lo que indica el buen comportamiento general del modelo frente a la clasificación de créditos en las dos categorías de preaprobación. Se destacan la espcificidad y la precisión las cuales lograron valores en promedio por encima del 77%, lo que supera el limite inferior definido para este tipo de modelos de clasificación el cual se ubica en el 75%